src/main/java/se/kb/libris/util/marc/io/Iso2709Serializer.java

```diff
@@ -209,4 +186,95 @@
         
         return rec;
     }
+    
+    private static byte[] serialize(Datafield df) {
+        try {
+            return serialize(df, null);
+        } catch (UnsupportedEncodingException cannotHappen) {
+            // Never throws unless passed an encoding
+            throw new RuntimeException(cannotHappen);
+        }
+    }
+    
+    private static byte[] serialize(Datafield df, String encoding) throws UnsupportedEncodingException {
+        StringBuilder sb = new StringBuilder();
+
+        sb.append(df.getIndicator(0));
+        sb.append(df.getIndicator(1));
+
+        Iterator<Subfield> sfiter = df.iterator();
+        while (sfiter.hasNext()) {
+            Subfield sf = sfiter.next();
+
+            sb.append((char)MarcRecord.SUBFIELD_MARK);
+            sb.append(sf.getCode());
+            sb.append(sf.getData());
+        }
+        
+        byte[] bytes = str2bytes(sb.toString(), encoding);
+
+        if (bytes.length > MAX_FIELD_LEN) {
```

Comment:
Could theoretically be an infinite recursion if there is thousands of empty subfields...
StackOverflowError to the rescue